src/command/service-request.ts

```diff
@@ -27,3 +27,57 @@ export function getCreateRequestFee(
 export function unstakeRequestFee(api: ApiPromise, pair: any, requestId: string): any {
   return api.tx.serviceRequest.unstake(requestId).paymentInfo(pair);
 }
+
+export async function generateRequestId(api: ApiPromise, pair: any, country: string, region: string, city: string, category: string) {
+  const result = await api.tx.serviceRequest
+    .generateRequestid(country, region, city, category)
+    .signAndSend(pair, { nonce: -1 });
+
+  return result.toHuman();
+}
+
+export async function unstakeRequest (
+  api: ApiPromise, 
+  pair: any, 
+  requestId: string, 
+  callback? : () => void,
+) {
+  const unsub = await api.tx.serviceRequest
+    .unstake(requestId)
+    .signAndSend(pair, { nonce: -1 }, ({ events, status }) => successCallback(api, { events, status, callback }));
+  unsub();
+}
+
+export async function claimRequest(api: ApiPromise, pair: any, requestId: string, serviceId: string, testingPrice: string, qcPrice: string) {
+  const result = await api.tx.serviceRequest
+    .claimRequest(requestId, serviceId, testingPrice, qcPrice)
+    .signAndSend(pair, { nonce: -1 });
+  return result.toHuman();
+}
+
+export async function processRequest(
+  api: ApiPromise, 
+  pair: any, 
+  labId: string, 
+  requestId: string,
+  orderId: string,
+  dnaSampleTrackingId: string,
+  additionalStakingAmount: string,
+) {
+  const result = await api.tx.serviceRequest
+    .processRequest(labId, requestId, orderId, dnaSampleTrackingId, additionalStakingAmount)
+    .signAndSend(pair, { nonce: -1 });
+  return result.toHuman();
+}
+
+export async function finalizeRequest(
+  api: ApiPromise,
+  pair: any,
+  requestId: string,
+  testResultSuccess: string,
+) {
+  const result = await api.tx.serviceRequest
+    .finalizeRequest(requestId, testResultSuccess)
+    .signAndSend(pair, { nonce: -1 });
+  return result.toHuman();
+}
```

Comment:
Where are the `unsub` and `successCallback`?

Reply:
in customer ui repo, some function is not using unsub and callback in signAndSend.
i just write base on the customer ui repo.

Reply:
Every `command` needs a callback available. Because it can be set to optional. @rubenkristian